src/main/java/org/dcsa/core/events/repository/ShipmentEventRepository.java

```diff
@@ -13,4 +13,34 @@
 
     @Query("SELECT * FROM shipment_event a WHERE (:eventType IS NULL or a.event_type =:eventType)")
     Flux<ShipmentEvent> findShipmentEventsByFilters(@Param("eventType") EventType eventType);
+
+  @Query(
+      "SELECT DISTINCT td.transport_document_reference from transport_document td "
+          + "JOIN shipping_instruction si ON shipping_instruction_id = si.id "
+          + "JOIN cargo_item ci ON si.id = ci.shipping_instruction_id "
+          + "JOIN shipment s ON ci .shipment_id = s.id "
+          + "WHERE s.carrier_booking_reference = :carrierBookingRef")
+  Flux<String> findTransportDocumentRefsByCarrierBookingRef(String carrierBookingRef);
+
+  @Query(
+      "SELECT DISTINCT b.carrier_booking_reference FROM booking b "
+          + "JOIN shipment s ON b.carrier_booking_reference = s.carrier_booking_reference "
+          + "JOIN cargo_item ci ON s.id = ci.shipment_id "
+          + "JOIN shipping_instruction si ON ci.shipping_instruction_id = si.id "
+          + "JOIN transport_document td ON si.id = td.shipping_instruction_id "
+          + "WHERE td.transport_document_reference = :transportDocumentRef")
+  Flux<String> findCarrierBookingRefsByTransportDocumentRef(String transportDocumentRef);
+
+  @Query(
+      "SELECT DISTINCT td.transport_document_reference FROM transport_document td "
+          + "JOIN shipping_instruction si ON td.shipping_instruction_id  = si .id "
+          + "JOIN cargo_item ci ON si.id = ci.shipping_instruction_id "
+          + "WHERE ci.shipment_id = :shipmentId")
+  Flux<String> findTransportDocumentRefsByShipmentId(UUID shipmentId);
+
+  @Query(
+      "SELECT DISTINCT b.carrier_booking_reference FROM booking b "
+          + "JOIN shipment s ON b.carrier_booking_reference = s.carrier_booking_reference "
+          + "WHERE s.id = :shipmentId")
+  Flux<String> findCarrierBookingRefsByShipmentId(UUID shipmentId);
```

Comment:
```suggestion
          + "WHERE s.id = :shipmentID")
  Flux<String> findCarrierBookingRefsByShipmentID(UUID shipmentID);
```

Remember to update callers as well (not covered in my review)